hotham-editor/src/main.rs

```diff
@@ -0,0 +1,418 @@
+mod camera;
+mod input_context;
+
+use anyhow::{bail, Result};
+use ash::vk;
+
+use glam::Vec2;
+use hotham_editor_protocol::{responses, EditorServer, RequestType};
+use lazy_vulkan::{
+    find_memorytype_index, vulkan_context::VulkanContext, vulkan_texture::VulkanTexture, DrawCall,
+    LazyRenderer, LazyVulkan, SwapchainInfo, Vertex,
+};
+use log::{debug, info, trace};
+
+use std::time::Instant;
+use uds_windows::{UnixListener, UnixStream};
+use winit::{
+    event::{ElementState, Event, KeyboardInput, VirtualKeyCode, WindowEvent},
+    event_loop::ControlFlow,
+    platform::run_return::EventLoopExtRunReturn,
+};
+
+use crate::camera::Camera;
+
+/// Compile your own damn shaders! LazyVulkan is just as lazy as you are!
+static FRAGMENT_SHADER: &'_ [u8] = include_bytes!("shaders/triangle.frag.spv");
+static VERTEX_SHADER: &'_ [u8] = include_bytes!("shaders/triangle.vert.spv");
```

Comment:
These files are not checked in

Reply:
Ah! Good pickup.

Reply:
You may want to add a `.gitignore` with exceptions for these files. Checking in files that are ignored by git can lead to confusing stuff, eg. local changes are not picked up by git even though the file is tracked.

Reply:
Hrm, I've found that it's usually fine after you've explicitly added that file to the index, but I've added an exception to the gitignore just to be on the safe side